### PR TITLE
Audit action field optionality

### LINKED ISSUE #372: Audit action field optionality
# Environment

**Delta-rs version**: all

**Binding**: rust

***
# Bug

**What happened**:

Some of the fields are incorrectly marked as optional in action.rs

**What you expected to happen**:

Make sure optionality annotation matches with what's in delta spec and reference implementation: https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/actions/actions.scala

**How to reproduce it**:

**More details**:

Follow up from https://github.com/delta-io/delta-rs/pull/370

### PR DESCRIPTION
**Pull request by fvaleye**
# Description
Audit Action attributes using https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/actions/actions.scala file

# Related Issue(s)
closes #372

### CODE REVIEW

**Comment by fvaleye** [comment]
@houqp based on my investigation, tell me what you think about:
- Keeping optional fields in Actions and Metadata when there are null by default in delta (examples from Metadata: [names and description](https://github.com/delta-io/delta/blob/master/core/src/main/scala/org/apache/spark/sql/delta/actions/actions.scala#L372-L373)). 
`Option` will be used to avoid serialization/deserialization errors when reading the delta transaction log.
I encountered several issues when reading older delta table versions <1.0 (these fields were not in the transaction logs).

- Settting `created_time` in `Metadata` as Optional 
- Settting `deletion_timestamp` in `Remove` as Optional
- Setting the values of `data_change` and `extended_file_metadata` with their default values when reading delta table transaction log for Remove actions

**Comment by houqp** [comment]
Sorry for the late response @fvaleye , catch up on notifications after getting back from vacation :) 

What you proposed makes total sense to me :+1:

File: rust/src/action.rs [hunk 1]
```diff
@@ -537,7 +538,7 @@ pub struct Remove {
     /// The path of the file that is removed from the table.
     pub path: String,
     /// The timestamp when the remove was added to table state.
-    pub deletion_timestamp: DeltaDataTypeTimestamp,
+    pub deletion_timestamp: Option<DeltaDataTypeTimestamp>,
```
Comment (houqp) [inline rust/src/action.rs:541]: I think this is incorrectly annotated in the official delta spec, we should also send a PR to the main delta repo to correct it :)

**Review by houqp — APPROVED** [review]

**Comment by fvaleye** [comment]
> Sorry for the late response @fvaleye , catch up on notifications after getting back from vacation :)
> 
> What you proposed makes total sense to me 👍

@houqp Oh, no worries at all! I hope you enjoyed it 🏖️

File: rust/src/action.rs (hunk 1 above)
Comment (fvaleye) [reply rust/src/action.rs:541]: Good idea, will do!

Comment (fvaleye) [reply rust/src/action.rs:541]: https://github.com/delta-io/delta/pull/775

**Pull request merged**